package javax.microedition.media;

public interface Player extends Controllable {
    int UNREALIZED = 100, REALIZED = 200, PREFETCHED = 300, STARTED = 400, CLOSED = 0;
    void realize() throws MediaException;
    void prefetch() throws MediaException;
    void start() throws MediaException;
    void stop() throws MediaException;
    void deallocate();
    void close();
    int getState();
    void setLoopCount(int count);
    long getDuration();
    long getMediaTime();
    long setMediaTime(long t) throws MediaException;
    String getContentType();
}
